### PR TITLE
ipv6: Tolerate multiple routes to the same NAT Gateway

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: upup/pkg/fi/cloudup/awstasks/natgateway.go [hunk 1]
```diff
@@ -225,14 +225,18 @@ func findNatGatewayFromRouteTable(cloud awsup.AWSCloud, routeTable *RouteTable)
 					return nil, fmt.Errorf("Could not find '%s' tag from route table", awsup.TagClusterName)
 				}
 				filteredNatGateways := []*ec2.NatGateway{}
+				natGatewayIDsSeen := map[string]bool{}
 				for _, natGatewayID := range natGatewayIDs {
 					gw, err := findNatGatewayById(cloud, natGatewayID)
 					if err != nil {
 						return nil, err
 					}
```
Comment (hakman) [inline upup/pkg/fi/cloudup/awstasks/natgateway.go:234]: Nit: Would it be easier to read if we skip early?
```suggestion
					if natGatewayIDsSeen[*natGatewayID] {
						continue
					}
```

Comment (johngmyers) [reply upup/pkg/fi/cloudup/awstasks/natgateway.go]: Moved it even earlier.

**Review by hakman — APPROVED** [review]

**Pull request merged**